server/routes/profile.js

```diff
@@ -4,10 +4,13 @@ const protect = require('../middleware/auth');
 const {
   editProfile,
   loadProfile,
+  getProfileListings
 } = require('../controllers/profile');
 
 router.route('/edit').put(protect, editProfile);
 
 router.route('/load').get(protect, loadProfile);
 
-module.exports = router;
\ No newline at end of file
+router.route('/:availability/:location/list-profiles').get(getProfileListings);
```

Comment:
Let's resort to using query params (used to query db based on conditions) instead of path params (used to identify a resource)